students/mark/Session04/file_read_parse_lab.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+
+"""
+File reading and parsing
+
+Download this text file:
+
+students.txt
+
+In it, you will find a list of names and what programming languages they have used in the past.
+This may be similar to a list generated at the beginning of this class.
+
+Write a little script that reads that file, and generates a list of all the languages that have been used.
+
+What might be the best data structure to use to keep track of bunch of values without duplication?
+"""
+
+
+### Note: we could assume we know all the languages, however, we're instead going to assumet everyone capitalizes thier nickname.
+valid_language=(
+    "ansible",
+    "bash",
+    "c",
+    "c#",
+    "c++",
+    "db",
+    "erlang",
+    "fortran",
+    "java",
+    "javascript",
+    "lisp",
+    "matlab",
+    "mysql",
+    "nothing",
+    "perl",
+    "php",
+    "powershell",
+    "python",
+    "r",
+    "rex",
+    "ruby",
+    "shell",
+    "sql",
+    "typescript",
+    "vb",
+    "visualbasic",
+    )
+
+
+students_file="./students.txt"
+
+with open(students_file, 'r') as f:
+    read_data = f.readlines()
+
+languages={}    # dict of languages
+nick_names={}   # dict or nick_names
+
+# Delete the header on the file
+read_data.reverse() # invert list order
+read_data.pop()     # pop off the header
+read_data.reverse() # put list back in orig order
+
```

Comment:
Clever. Another way to do it is to read the first line when you open the file, and then the rest:
```
with open('students.txt', 'r') as f:
    first_line = f.readline()
    read_data = f.readlines()
```